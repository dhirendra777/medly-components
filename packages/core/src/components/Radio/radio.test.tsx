import { render } from '@test-utils';
import React from 'react';
import { Radio } from './Radio';

describe('Radio component', () => {
    it('should render correctly with all the default props', () => {
        const { container } = render(<Radio value="female" name="Female" />);
        expect(container).toMatchSnapshot();
    });

    it('should render correctly with all the props given', () => {
        const { container } = render(
            <Radio disabled fullWidth name="gender" value="female" label="Female" size="M" labelPosition="right" />
        );
        expect(container).toMatchSnapshot();
    });
});
