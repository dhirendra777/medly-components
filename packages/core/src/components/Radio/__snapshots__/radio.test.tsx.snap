// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio component should render correctly with all the default props 1`] = `
.c0 {
  margin: 8px 8px 8px 0px;
  display: inline-grid;
  grid-template-columns: max-content minmax(max-content,25%);
  grid-template-areas: 'field label' 'description random';
  grid-template-columns: auto;
  grid-template-areas: 'field' 'description';
}

.c0 .c5 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
}

.c4 {
  background-color: #012040;
  width: 0;
  height: 0;
  z-index: 1;
  border-radius: 100%;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  opacity: 0;
  z-index: 2;
  position: absolute;
  width: 100%;
  height: 100%;
  margin: 0;
  cursor: pointer;
}

.c2:checked ~ .c3 {
  width: calc(75%);
  height: calc(75%);
  -webkit-transition: width 0.2s ease,height 0.2s ease;
  transition: width 0.2s ease,height 0.2s ease;
}

.c1 {
  width: 16px;
  height: 16px;
  position: relative;
  border-radius: 100%;
  border: 1px solid #EEEEEE;
  background-color: #ffffff;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1:focus-within {
  box-shadow: 0 0 0 1pt #012040;
}

<div>
  <div
    class="c0"
    id="-radio"
  >
    
    <div
      class="c1"
    >
      <input
        class="c2"
        id=""
        name="Female"
        type="radio"
        value="female"
      />
      <div
        class="c3 c4"
      />
    </div>
  </div>
</div>
`;

exports[`Radio component should render correctly with all the props given 1`] = `
.c1 {
  font-size: 14px;
  font-weight: 400;
  color: #616161;
  margin: 0px 0px 0px 8px;
  cursor: not-allowed;
}

.c2 {
  grid-area: label;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c0 {
  margin: 8px 0px;
  display: grid;
  grid-template-columns: max-content minmax(max-content,25%);
  grid-template-areas: 'field label' 'description random';
}

.c0 .c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
}

.c6 {
  background-color: #012040;
  width: 0;
  height: 0;
  z-index: 1;
  border-radius: 100%;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c7:checked ~ .c5 {
  width: calc(75%);
  height: calc(75%);
  -webkit-transition: width 0.2s ease,height 0.2s ease;
  transition: width 0.2s ease,height 0.2s ease;
}

.c4 {
  opacity: 0;
  z-index: 2;
  position: absolute;
  width: 100%;
  height: 100%;
  margin: 0;
  cursor: not-allowed;
}

.c4:checked ~ .c5 {
  width: calc(75%);
  height: calc(75%);
  -webkit-transition: width 0.2s ease,height 0.2s ease;
  transition: width 0.2s ease,height 0.2s ease;
}

.c3 {
  width: 24px;
  height: 24px;
  position: relative;
  border-radius: 100%;
  border: 1px solid #EEEEEE;
  background-color: #E0E0E0;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3:focus-within {
  box-shadow: 0 0 0 1pt #012040;
}

<div>
  <div
    class="c0"
    id="Female-radio"
  >
    <label
      class="c1 c2"
      for="Female"
    >
      Female
    </label>
    <div
      class="c3"
      disabled=""
    >
      <input
        class="c4"
        disabled=""
        id="Female"
        name="gender"
        type="radio"
        value="female"
      />
      <div
        class="c5 c6"
      />
    </div>
  </div>
</div>
`;
